tests/unit/resources/settings/test_restores.py

```diff
@@ -20,70 +20,26 @@
+from hpOneView.resources.resource import Resource, ResourceHelper
 from hpOneView.resources.settings.restores import Restores
 
 
 class RestoresTest(TestCase):
     def setUp(self):
-        oneview_connection = connection('127.0.0.1')
-        self.resource = Restores(oneview_connection)
+        self.host = '127.0.0.1'
+        self.connection = connection(self.host)
+        self.resource = Restores(self.connection)
 
-        self.restores = [
-            {"backupIdToRestore": "example_backup_2017-03-06_023131"},
-            {"backupIdToRestore": "example_backup_2017-04-06_023131"}
-        ]
-
-    @mock.patch.object(ResourceClient, 'get_all')
-    def test_get_all(self, mock_get_all):
-        self.resource.get_all()
-        mock_get_all.assert_called_once()
-
-    @mock.patch.object(ResourceClient, 'get_all')
-    def test_get_by_restores_when_found(self, mock_get_all):
-        mock_get_all.return_value = self.restores
-
-        expected_result = [self.restores[0]]
-
-        result = self.resource.get_by("backupIdToRestore", "example_backup_2017-03-06_023131")
-
-        mock_get_all.assert_called_once_with(filter='"backupIdToRestore=\'example_backup_2017-03-06_023131\'"',
-                                             uri='/rest/restores')
-        self.assertEqual(result, expected_result)
-
-    @mock.patch.object(ResourceClient, 'get_all')
-    def test_get_by_should_return_empty_list_when_not_found(self, mock_get_all):
-        mock_get_all.return_value = []
-
-        result = self.resource.get_by("backupIdToRestore", "example_backup_2017-03-06_023131")
-
-        self.assertEqual(result, [])
-
-    @mock.patch.object(ResourceClient, 'get')
-    def test_get_should_be_called_once(self, mock_get):
-        self.resource.get('3518be0e-17c1-4189-8f81-83f3724f6155')
-
-        mock_get.assert_called_once_with('3518be0e-17c1-4189-8f81-83f3724f6155')
-
-    @mock.patch.object(ResourceClient, 'get')
-    def test_get_with_uri_should_be_called_once(self, mock_get):
-        uri = '/rest/restores/3518be0e-17c1-4189-8f81-83f3724f6155'
-        self.resource.get(uri)
-
-        mock_get.assert_called_once_with(uri)
-
-    @mock.patch.object(ResourceClient, 'get')
+    @mock.patch.object(ResourceHelper, 'do_get')
     def test_get_failure_should_be_called_once(self, mock_get):
         uri = '/rest/restores/failure'
         self.resource.get_failure()
-
         mock_get.assert_called_once_with(uri)
 
-    @mock.patch.object(ResourceClient, 'create')
+    @mock.patch.object(Resource, 'create')
     def test_restore_should_be_called_once(self, mock_create):
         restore = {
             "uriOfBackupToRestore": "/rest/backups/example_backup_2014-03-06_023131"
         }
         self.resource.restore(restore)
-
-        mock_create.assert_called_once_with(restore, timeout=-1, default_values=self.resource.DEFAULT_VALUES)
+        mock_create.assert_called_once_with(restore, timeout=-1)
```

Comment:
Why arent we using default values here, any reason to it?

Reply:
this refactored code is using the new base class (Resource) and in that class we are handling those default_values. So need to pass them in resource library.